proto/redpanda/core/admin/cluster_link.proto

```diff
@@ -0,0 +1,407 @@
+    // The UUID of the cluster link
+    string uid = 2 [
+        (google.api.field_behavior) = OUTPUT_ONLY,
+        (google.api.field_info).format = UUID4
+    ];
+    // Cluster link configuration
+    ClusterLinkConfiguration configuration = 3;
+    // Status of the cluster link
+    ClusterLinkStatus status = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+
+/* RPC Requests and Responses */
+// Create a new cluster link
+message CreateClusterLinkRequest {
+    // The cluster link to create
+    ClusterLink cluster_link = 1;
+}
+// Request to delete a cluster link
+message DeleteClusterLinkRequest {
+    // The name of the link to delete
+    string name = 1 [
+        (google.api.field_behavior) = REQUIRED,
+        (google.api.resource_reference) = {
+            type: "redpanda.core.admin.ClusterLinkService/ClusterLink"
+        }
+    ];
+}
+message DeleteClusterLinkResponse {}
+message GetClusterLinkRequest {
+    // The name of the cluster link to get
+    string name = 1 [
+        (google.api.field_behavior) = REQUIRED,
+        (google.api.resource_reference) = {
+            type: "redpanda.core.admin.ClusterLinkService/ClusterLink"
+        }
+    ];
+}
+// Request to list all cluster links
+message ListClusterLinksRequest {}
+// All cluster links on the cluster
+message ListClusterLinksResponse {
+    // The cluster links
+    repeated ClusterLink cluster_links = 1;
+}
+// Updates a cluster link
+message UpdateClusterLinkRequest {
+    // The cluster link to update
+    ClusterLink cluster_link = 1;
+
+    // The list of fields to update
+    // See [AIP-134](https://google.aip.dev/134) for how to use `field_mask`
+    google.protobuf.FieldMask update_mask = 2;
+}
+// Request to fail over a cluster link or a single mirror topic
+message FailOverRequest {
+    // The name of the cluster link to fail over
+    string name = 1 [
+        (google.api.field_behavior) = REQUIRED,
+        (google.api.resource_reference) = {
+            type: "redpanda.core.admin.ClusterLinkService/ClusterLink"
+        }
+    ];
+    // The name of the mirror topic to fail over, if not set will fail over
+    // the entire cluster link
+    string mirror_topic_name = 2 [(google.api.field_behavior) = OPTIONAL];
+}
+
+/* Other Messages */
+// ClusterLink options
+message ClusterLinkConfiguration {
+    // Configuration for the internal kafka client
+    ClusterLinkClientOptions client_options = 1;
+    // Topic metadata sync options
+    TopicMetadataSyncOptions topic_metadata_sync_options = 2;
+    // Consumer offset sync options
+    ConsumerOffsetSyncOptions consumer_offset_sync_options = 3;
+    // Security settings sync options
+    SecuritySettingsSyncOptions security_sync_options = 4;
+}
+// Options for the client link
+message ClusterLinkClientOptions {
+    // The bootstrap servers to use
+    repeated string bootstrap_servers = 1
+        [(google.api.field_behavior) = REQUIRED];
+    // The Client ID
```

Comment:
```suggestion
    // The Client ID for the Kafka RPC requests sent by this cluster link to the source cluster. 
```

Reply:
While obvious to us, mere mortals who configure this stuff may not be in the same headspace :)